### PR TITLE
A bunch of changes to fix creation of VCD files by interpreter

### PR DESCRIPTION
**Pull request by chick**
Add wires to scopes when they are first encountered
Update a mythical clock signal as interpreter cycles
Add scopes from wires as they are encountered
Add instance name tracking to dependency graph
Removed requirement that repl only allows vcd output if script is loaded
repl now shows null pointer exceptions without dying
Add fft firrtl file to resources

### CODE REVIEW

File: src/main/scala/firrtl_interpreter/CircuitState.scala [hunk 1]
```diff
@@ -116,19 +123,25 @@ case class CircuitState(
...
-    vcdLoggerOption.foreach { vcd =>
-      vcd.incrementTime()
-    }
     nameToConcreteValue = mutable.HashMap((inputPorts ++ outputPorts ++ registers).toSeq:_*)
     isStale = true
 
     stateCounter += 1
+
+    vcdLoggerOption.foreach { vcd =>
+      vcd.incrementTime(10)
```
Comment (kammoh) [inline src/main/scala/firrtl_interpreter/CircuitState.scala]: I guess [https://github.com/ucb-bar/firrtl-interpreter/blob/38cf05bee320d9802f8d662cc02315a3d7047607/src/main/scala/firrtl_interpreter/vcd/VCD.scala#L541](url) needs to be fixed to `timeStamp += increment`

**Comment by grebe** [comment]
This is working for me with the latest and greatest fft (except that sometimes an assertion stops execution, presumably b/c I've got some initial state wrong somewhere).

I do notice that the VCD hierarchy is only 1 module deep. Also, the names of signals in modules that are not the top module are a little funny- they have a \ in front and have the module name at the beginning. It would be nice if the names were a little cleaner, but it is certainly functional.

I don't have any code quibbles.

Thanks, this is awesome!

**Comment by grebe** [comment]
I suppose this PR should reference #35

File: src/main/scala/firrtl_interpreter/CircuitState.scala (hunk 1 above)
Comment (chick) [inline src/main/scala/firrtl_interpreter/CircuitState.scala]: @kammoh : good catch, thanks

Comment (kammoh) [inline src/main/scala/firrtl_interpreter/CircuitState.scala]: yw;)
actually I'd made kinda similar patch yesterday, including added hierarchy to 'Scope's, but was facing some problems with register updates. I see a similar issue in this PR as well. I just tried the VendingMachine example and the "state" register appears to be showing nextstate istead of it's current state in the vcd. One more thing, the first timestamp in the vcd is -1 which my vcd viewer doesn't seem to like. Not quite sure what the vcd spec says about this.

Comment (chick) [inline src/main/scala/firrtl_interpreter/CircuitState.scala]: I'm working on patching up the scoping stuff a bit more this afternoon.  I'll think about the -1, I was using that a shorthand for initial state, but I can probably pick something more reasonable

**Pull request merged**